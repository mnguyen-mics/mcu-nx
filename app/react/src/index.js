import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { Navigator } from './containers/Navigator';

const store = configureStore();

const micsProvider = (
  <Provider store={store}>
    <Navigator />
  </Provider>
);

render(
  micsProvider,
  document.getElementById('mcs-react-app') // eslint-disable-line no-undef
);
