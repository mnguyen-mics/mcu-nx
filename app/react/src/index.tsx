/* eslint-disable */
import React from 'react';
import { render } from 'react-dom';

import App from './App';

declare global {
  interface Window {
    require: any;
    mainRev: any;
  }
}

render(<App />, document.getElementById('mcs-react-app'), () => {
  window.require([window.mainRev], () => {
    //
  });
});
