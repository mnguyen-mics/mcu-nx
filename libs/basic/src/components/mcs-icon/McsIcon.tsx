import { HomeOutlined } from '@ant-design/icons';
import React from 'react';

export type McsIconType =
  | 'adGroups'
  | 'ads'
  | 'automation'
  | 'bell'
  | 'bolt'
  | 'check-rounded-inverted'
  | 'check-rounded'
  | 'check'
  | 'chevron-right'
  | 'chevron'
  | 'close-big'
  | 'close-rounded'
  | 'close'
  | 'code'
  | 'creative'
  | 'data'
  | 'delete'
  | 'display'
  | 'dots'
  | 'download'
  | 'email-inverted'
  | 'email'
  | 'extend'
  | 'filters'
  | 'full-users'
  | 'file'
  | 'gears'
  | 'goals-rounded'
  | 'goals'
  | 'home'
  | 'image'
  | 'info'
  | 'laptop'
  | 'library'
  | 'magnifier'
  | 'menu-close'
  | 'minus'
  | 'optimization'
  | 'options'
  | 'partitions'
  | 'pause'
  | 'pen'
  | 'phone'
  | 'play'
  | 'plus'
  | 'query'
  | 'question'
  | 'refresh'
  | 'settings'
  | 'smartphone'
  | 'status'
  | 'tablet'
  | 'user'
  | 'users'
  | 'user-query'
  | 'user-pixel'
  | 'user-list'
  | 'video'
  | 'warning';

const antdIcons: McsIconType[] = ['home'];

export interface McsIconProps {
  type: McsIconType;
  styleIcon?: React.CSSProperties;
}

class McsIcon extends React.Component<
  McsIconProps & React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>
> {
  render() {
    const { type, styleIcon, className, ...rest } = this.props;

    if (antdIcons.includes(type)) {
      if (type === 'home') return <HomeOutlined className={'mcs-icon'} style={rest.style} />;
      else return <span />;
    } else
      return (
        <span className={`mcs-icon ${className ? className : ''}`} {...rest}>
          <i className={`mcs-${type}`} style={styleIcon ? styleIcon : {}} />
        </span>
      );
  }
}

export default McsIcon;
