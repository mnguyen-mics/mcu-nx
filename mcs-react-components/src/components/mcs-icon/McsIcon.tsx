import * as React from 'react';

export type McsIconType =
  'adGroups' |
  'ads' |
  'automation' |
  'bell' |
  'bolt' |
  'check-rounded-inverted' |
  'check-rounded' |
  'check' |
  'chevron-right' |
  'chevron' |
  'close-big' |
  'close-rounded' |
  'close' |
  'code' |
  'creative' |
  'data' |
  'delete' |
  'display' |
  'dots' |
  'download' |
  'email-inverted' |
  'email' |
  'extend' |
  'filters' |
  'full-users' |
  'file' |
  'gears' |
  'goals-rounded' |
  'goals' |
  'image' |
  'info' |
  'laptop' |
  'library' |
  'magnifier' |
  'menu-close' |
  'minus' |
  'optimization' |
  'options' |
  'partitions' |
  'pause' |
  'pen' |
  'phone' |
  'play' |
  'plus' |
  'query' |
  'question' |
  'refresh' |
  'settings' |
  'smartphone' |
  'status' |
  'tablet' |
  'user' |
  'users' |
  'user-query' |
  'user-pixel' |
  'user-list' |
  'video' |
  'warning';

export interface McsIconProps {
  type: McsIconType;
  styleIcon?: React.CSSProperties;
}

class McsIcon extends React.Component<
  McsIconProps &
    React.DetailedHTMLProps<
      React.HTMLAttributes<HTMLSpanElement>,
      HTMLSpanElement
    >
> {
  render() {
    const { type, styleIcon, className, ...rest } = this.props;
    return (
      <span className={`mcs-icon ${className ? className : ''}`} {...rest}>
        <i className={`mcs-${type}`} style={styleIcon ? styleIcon : {}}/>
      </span>
    );
  }
}

export default McsIcon;