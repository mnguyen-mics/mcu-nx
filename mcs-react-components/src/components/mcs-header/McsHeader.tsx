import * as React from 'react';
import { AppstoreFilled } from '@ant-design/icons';
import { Layout, Menu, Dropdown } from 'antd';
import McsIcon from '../mcs-icon';

const { Header } = Layout;

export interface McsHeaderProps {
  menu?: React.ReactElement;
  menuIcon?: React.ReactElement;
  userEmail: string;
  accountContent: React.ReactNode[];
  headerSettings?: React.ReactNode;
  devAlert?: React.ReactNode;
}

class McsHeader extends React.Component<McsHeaderProps> {
  render() {
    const {
      menu,
      userEmail,
      accountContent,
      headerSettings,
      devAlert,
      menuIcon,
    } = this.props;

    const accountMenu = (
      <Menu>
        <Menu.Item key="email" disabled={true}>
          {userEmail}
        </Menu.Item>
        {accountContent.length > 0 && <Menu.Divider />}
        {accountContent.map((item, index) => {
          return <Menu.Item key={index}>{item}</Menu.Item>;
        })}
      </Menu>
    );

    return (
      <Header className="mcs-header">
        <div className="mcs-header-wrapper">
          <span className="mcs-header-left-component">
            {menu && (
              <span className="mcs-header-wrapper-launcher">
                <Dropdown overlay={menu} trigger={['click']}>
                  <a>
                    {menuIcon ? (
                      menuIcon
                    ) : (
                      <AppstoreFilled className="mcs-header-menu-icon" />
                    )}
                  </a>
                </Dropdown>
              </span>
            )}
          </span>
          {devAlert}
        </div>
        <div className="mcs-header-actions">
          {headerSettings && (
            <div className="mcs-header-actions-settings">{headerSettings}</div>
          )}
          <div className="mcs-header-actions-account">
            <Dropdown
              overlay={accountMenu}
              trigger={['click']}
              placement="bottomRight"
            >
              <a>
                <McsIcon type="user" className="mcs-header-menu-icon" />
              </a>
            </Dropdown>
          </div>
        </div>
      </Header>
    );
  }
}

export default McsHeader;
