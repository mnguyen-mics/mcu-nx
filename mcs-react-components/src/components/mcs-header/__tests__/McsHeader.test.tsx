import { Alert, Menu } from 'antd';
import 'jest';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import McsIcon from '../../mcs-icon';
import McsHeader, { McsHeaderProps } from '../McsHeader';

it('renders the Error component', () => {
  const props: McsHeaderProps = {
    headerTitle: 'Mediarithmics New App',
    userEmail: 'toto@mix.com',
    devAlert: (
      <Alert
        style={{
          display: 'flex',
          margin: 'auto',
          marginRight: 0,
          borderColor: 'red',
        }}
        message="You are using production API environment !"
        type="error"
        showIcon={true}
      />
    ),
    headerSettings: (
      <a>
        <McsIcon
          type="options"
          className="mcs-header-menu-icon"
          style={{ display: 'flex' }}
        />
      </a>
    ),
    menu: (
      <Menu>
        <Menu.Item>Completely optionnal</Menu.Item>
        <Menu.Item>Really optionnal</Menu.Item>
      </Menu>
    ),
    /* tslint:disable */
    accountContent: [<div>Account Menu #1</div>, <div>Account Menu #2</div>],
    /* tslint:enable */
  };

  const component = TestRenderer.create(<McsHeader {...props} />);
  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
