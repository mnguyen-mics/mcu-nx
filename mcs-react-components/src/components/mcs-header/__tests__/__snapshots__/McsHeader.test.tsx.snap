// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Error component 1`] = `
<header
  className="ant-layout-header mcs-header"
>
  <div
    className="mcs-header-wrapper"
  >
    <span
      className="mcs-header-left-component"
    >
      <span
        className="mcs-header-wrapper-launcher"
      >
        <a
          className="ant-dropdown-trigger"
          onClick={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
        >
          <span
            aria-label="appstore"
            className="anticon anticon-appstore mcs-header-menu-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="appstore"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M864 144H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm0 400H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zM464 144H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm0 400H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16z"
              />
            </svg>
          </span>
        </a>
      </span>
      <span
        className="mcs-header-wrapper-title"
      >
        Mediarithmics New App
      </span>
    </span>
    <div
      className="ant-alert ant-alert-error"
      data-show={true}
      role="alert"
      style={
        Object {
          "borderColor": "red",
          "display": "flex",
          "margin": "auto",
          "marginRight": 0,
        }
      }
    >
      <span
        aria-label="close-circle"
        className="anticon anticon-close-circle ant-alert-icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="close-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
          />
        </svg>
      </span>
      <div
        className="ant-alert-content"
      >
        <div
          className="ant-alert-message"
        >
          You are using production API environment !
        </div>
        <div
          className="ant-alert-description"
        />
      </div>
    </div>
  </div>
  <div
    className="mcs-header-actions"
  >
    <div
      className="mcs-header-actions-settings"
    >
      <a>
        <span
          className="mcs-icon mcs-header-menu-icon"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <i
            className="mcs-options"
            style={Object {}}
          />
        </span>
      </a>
    </div>
    <div
      className="mcs-header-actions-account"
    >
      <a
        className="ant-dropdown-trigger"
        onClick={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="mcs-icon mcs-header-menu-icon"
        >
          <i
            className="mcs-user"
            style={Object {}}
          />
        </span>
      </a>
    </div>
  </div>
</header>
`;
